core/src/main/java/org/apache/calcite/util/graph/DefaultDirectedGraph.java

```diff
@@ -36,7 +36,17 @@
 public class DefaultDirectedGraph<V, E extends DefaultEdge>
     implements DirectedGraph<V, E> {
   final Set<E> edges = new LinkedHashSet<>();
-  final Map<V, VertexInfo<V, E>> vertexMap = new LinkedHashMap<>();
+
+  /**
+   * The map for out-edges.
+   */
+  final Map<V, VertexInfo<V, E>> vertexOutMap = new LinkedHashMap<>();
```

Comment:
Why is VertexInfo used as Map value? Essentially you just need List\<E\>, don't you?

Reply:
I have removed VertexInfo. Thanks a lot for the good suggestion.